logicrunner/handle_calls.go

```diff
@@ -133,12 +135,22 @@ func (h *HandleCall) handleActual(
 	request := msg.IncomingRequest
 
 	procRegisterRequest := NewRegisterIncomingRequest(request, h.dep)
-	if err := f.Procedure(ctx, procRegisterRequest, true); err != nil {
+	err := f.Procedure(ctx, procRegisterRequest, true)
+	if err != nil {
 		if err == flow.ErrCancelled {
 			inslogger.FromContext(ctx).Info("pulse change during registration, asking caller for retry")
 			// Requests need to be deduplicated. For now in case of ErrCancelled we may have 2 registered requests
 			return nil, err // message bus will retry on the calling side in ContractRequester
 		}
+		if strings.Contains(err.Error(), "index not found") {
```

Comment:
here we can check errors.Cause by type/interface or whatever instead of Contains. strings.Contains should be used only when type is lost (after json/cbor desialization)

Reply:
not going to do it in this pull request, going to create a task

Reply:
alright, this happens after serialization/deserialization. So we can only check strings :(